src/compiler/checker.ts

```diff
@@ -13057,19 +13057,16 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             return;
         }
         // Combinations of function, class, enum and module
-        let members = emptySymbols;
+        let members = getExportsOfSymbol(symbol);
         let indexInfos: IndexInfo[] | undefined;
-        if (symbol.exports) {

```

Comment:
From what I understand `symbol.exports` only are only meant to keep properties that pass `isTypeUsableAsPropertyName` test and this symbol table is lazily created. Since symbols don't pass this test there was a chance for the `symbol.exports` table to not be created at all.

This patch essentially just grabs `getExportsOfSymbol` earlier.

Reply:
There is also still a question if this should be an error in JS files - since that PR doesn't change the behavior in JS. I'm not sure what's expected here though since both expando functions and JS files have their own unique rules

Reply:
The same code should work in JS. I'm guessing this change does that--can you add a test for it?

Reply:
I added the JS-oriented test case (with types declared in a TS file since I have no idea how to declare a computed property in JSDoc). I also have no idea what I had in mind here when posting that comment - this code here isn't targeting TS files (or non-JS files or anything like that) 😅